src/deephaven/plugin/js.py

```diff
@@ -6,29 +6,44 @@
 
 
 class JsPlugin(Plugin):
-    """A js plugin. Useful for adding custom JS code to the server that can be passed to the web client."""
+    """
+    A JS plugin is a Plugin that allows adding javascript code under the server's URL path "js-plugins/".
+    See https://github.com/deephaven/deephaven-plugins#js-plugins for more details about the underlying
+    construction for JS plugins.
+    """
 
     @abc.abstractmethod
-    def distribution_path(self) -> typing.Generator[pathlib.Path, None, None]:
-        """A generator that yields the distribution path directory."""
+    def root(self) -> typing.Generator[pathlib.Path, None, None]:
+        """
+        The root directory of the resources to serve. The root must exist.
+        """
```

Comment:
Any reason to prefer `root` over `path`? Feel like `def path` would be the more natural name.

Reply:
I don't have a strong opinion. `root`, `path`, `root_path` all seem reasonable to me. @niloc132? Either way, I think we should mirror this on the java side; `root`, `path`, or `rootPath`.

Reply:
My only pushback against `path` would be that "path" is already implied by the return type.

Reply:
I'd lean toward (but not block on) making path part of the name yeah - even in Java I think I'd prefer that, but doubly so in python where types not even strong suggestions...

Reply:
Will do.